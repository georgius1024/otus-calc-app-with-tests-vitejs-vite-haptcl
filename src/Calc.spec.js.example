import { describe, it, expect, beforeAll, beforeEach, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import component from './Calc.vue';

describe('Calculator', () => {
  const buildWrapper = () => {
    return mount(component);
  };
  it('mounts without error', () => {
    const wrapper = buildWrapper();
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('[data-testid="display"]').text()).toBe('0');
  });

  it('all digit buttons works', async () => {
    const wrapper = buildWrapper();
    await wrapper.find('[data-testid="btn-1"]').trigger('click');
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-3"]').trigger('click');
    await wrapper.find('[data-testid="btn-4"]').trigger('click');
    await wrapper.find('[data-testid="btn-5"]').trigger('click');
    await wrapper.find('[data-testid="btn-6"]').trigger('click');
    await wrapper.find('[data-testid="btn-7"]').trigger('click');
    await wrapper.find('[data-testid="btn-8"]').trigger('click');
    await wrapper.find('[data-testid="btn-9"]').trigger('click');
    await wrapper.find('[data-testid="btn-0"]').trigger('click');
    expect(wrapper.find('[data-testid="display"]').text()).toBe('1234567890');
  });

  it('can add two numbers', async () => {
    const wrapper = buildWrapper();
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-+"]').trigger('click');
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-="]').trigger('click');
    expect(wrapper.find('[data-testid="display"]').text()).toBe('4');
  });

  it('can multiply two numbers', async () => {
    const wrapper = buildWrapper();
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-*"]').trigger('click');
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-="]').trigger('click');
    expect(wrapper.find('[data-testid="display"]').text()).toBe('4');
  });

  it('can divide two numbers', async () => {
    const wrapper = buildWrapper();
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-/"]').trigger('click');
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-="]').trigger('click');
    expect(wrapper.find('[data-testid="display"]').text()).toBe('1');
  });

  it('can subtract two numbers', async () => {
    const wrapper = buildWrapper();
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn--"]').trigger('click');
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-="]').trigger('click');
    expect(wrapper.find('[data-testid="display"]').text()).toBe('0');
  });

  it('can clear inut', async () => {
    const wrapper = buildWrapper();
    await wrapper.find('[data-testid="btn-2"]').trigger('click');
    await wrapper.find('[data-testid="btn-C"]').trigger('click');
    expect(wrapper.find('[data-testid="display"]').text()).toBe('0');
  });
});
