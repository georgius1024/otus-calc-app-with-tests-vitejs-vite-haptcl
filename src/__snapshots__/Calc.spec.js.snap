// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Calculator > mounts without error 1`] = `
"<div class="App">
  <div class="display" data-testid="display">0</div>
  <div class="row"><button data-testid="btn-1">1</button><button data-testid="btn-2">2</button><button data-testid="btn-3">3</button><button data-testid="btn-*">*</button></div>
  <div class="row"><button data-testid="btn-4">4</button><button data-testid="btn-5">5</button><button data-testid="btn-6">6</button><button data-testid="btn-/">/</button></div>
  <div class="row"><button data-testid="btn-7">7</button><button data-testid="btn-8">8</button><button data-testid="btn-9">9</button><button data-testid="btn-+">+</button></div>
  <div class="row"><button data-testid="btn-=">=</button><button data-testid="btn-C">C</button><button data-testid="btn-0">0</button><button data-testid="btn--">-</button></div>
</div>"
`;
